api/custom-map-ui/geoname-search-ui.html

```diff
@@ -427,35 +460,36 @@
         // displayResults: JSON -> None
         // Effects: Populates the results container with search results
         function displayResults(obj) {
+            // Reset the results
+            document.querySelector("#results").innerHTML = "";
+            
             // Enable display panel
             document.getElementById("results").style.display = "block";
 
             for (let i=0; i < obj.length; i++) {
-                document.querySelector("#results").innerHTML = document.querySelector("#results").innerHTML + "<div class='result' role='button' tabindex='0' onkeypress='handleEnter(event)' onclick='mapPane(" + i + ")'>" + obj[i].name + "<div class='result_description'>" + obj[i].concise.code + " (" + obj[i].decision + ")</div></div>";
+                document.querySelector("#results").innerHTML = document.querySelector("#results").innerHTML + "<div id = 'item" + i + "' class='result' role='row' tabindex='0' onkeypress='handleEnter(event)' onclick='mapPane(" + i + ")'>" + obj[i].name + "<div class='result_description'>" + obj[i].concise.code + " (" + obj[i].decision + ")</div></div>";
```

Comment:
On the `role="row"` usage:

https://www.w3.org/TR/wai-aria-1.1/#row:

> Authors **_MUST_** ensure elements with role `row` are contained in, or owned by, an element with the role `table`, `grid`, `rowgroup`, or `treegrid`.

Currently does not meet the above requirements. However, `role="row"` doesn't indicate to screen reader users that these can be activated, I believe `role="button"` should suffice.

---

Side note: there's an established ARIA design pattern called [Combo Box](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox) that you may find informative (see [examples](https://www.w3.org/TR/wai-aria-practices-1.1/#examples-1)), I'm not suggesting that you rewrite this component but you may find it useful in the future.

Reply:
Using role="button" was not working using a screen reader after the  changes I made (lines 146-173) to incorporate arrow keys to navigate the results. I have updated the code such that role="row" elements are now children of row="grid". 

Combo Box looks interesting also. Thanks!